web/secondary_toolbar.js

```diff
@@ -29,173 +29,228 @@
 var SCROLLBAR_PADDING = uiUtils.SCROLLBAR_PADDING;
 var mozL10n = uiUtils.mozL10n;
 
-var SecondaryToolbar = {
-  opened: false,
-  previousContainerHeight: null,
-  newContainerHeight: null,
+/**
+ * @typedef {Object} SecondaryToolbarOptions
+ * @property {HTMLDivElement} toolbar - Container for the secondary toolbar.
+ * @property {HTMLButtonElement} toggleButton - Button for toggling the
+ *   visibility of the secondary toolbar.
+ * @property {HTMLButtonElement} presentationModeButton - Button for toggling
+ *   presentation mode.
+ * @property {HTMLButtonElement} openFile - Button for opening a file.
+ * @property {HTMLButtonElement} print - Button for printing the document.
+ * @property {HTMLButtonElement} download - Button for downloading the document.
+ * @property {HTMLLinkElement} viewBookmark - Button for obtaining a bookmark
+ *   link to the current location in the document.
+ * @property {HTMLButtonElement} firstPage - Button for going to the first page
+ *   in the document.
+ * @property {HTMLButtonElement} lastPage - Button for going to the last page
+ *   in the document.
+ * @property {HTMLButtonElement} pageRotateCw - Button for rotating the page
+ *   clockwise.
+ * @property {HTMLButtonElement} pageRotateCcw - Button for rotating the page
+ *   counterclockwise.
+ * @property {HTMLButtonElement} toggleHandTool - Button for toggling the hand
+ *   tool.
+ * @property {HTMLButtonElement} documentPropertiesButton - Button for toggling
+ *   the visibility of a document properties overlay.
+ */
 
-  initialize: function secondaryToolbarInitialize(options, eventBus) {
-    this.eventBus = eventBus;
+/**
+ * @class
+ */
+var SecondaryToolbar = (function SecondaryToolbarClosure() {
+  /**
+   * @constructs SecondaryToolbar
+   * @param {SecondaryToolbarOptions} options
+   * @param {EventBus} eventBus
+   */
+  function SecondaryToolbar(options, eventBus) {
     this.toolbar = options.toolbar;
-    this.buttonContainer = this.toolbar.firstElementChild;
-
-    // Define the toolbar buttons.
     this.toggleButton = options.toggleButton;
     this.presentationModeButton = options.presentationModeButton;
```

Comment:
This is a minor (existing) inconsistency, where some of the options have `...Button` on the end, and others don't. This is likely to be for "historical reasons", but since you're touching this code anyway, it'd be nice to make this more consistent :-)